hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1873,18 +1874,27 @@ public List<OzoneFileStatus> listStatusFSO(OmKeyArgs args, boolean recursive,
   private List<OzoneFileStatus> buildFinalStatusList(
       Map<String, OzoneFileStatus> cacheFileMap,
       Map<String, OzoneFileStatus> cacheDirMap, OmKeyArgs omKeyArgs,
-      String clientAddress)
+      String clientAddress, long numEntries)
       throws IOException {
     List<OmKeyInfo> keyInfoList = new ArrayList<>();
     List<OzoneFileStatus> fileStatusFinalList = new ArrayList<>();
 
+    long countEntries = 0;
     for (OzoneFileStatus fileStatus : cacheFileMap.values()) {
       fileStatusFinalList.add(fileStatus);
       keyInfoList.add(fileStatus.getKeyInfo());
+      countEntries++;
+      if (countEntries >= numEntries) {
```

Comment:
Do the keys in the `fileStatusFinalList` be a sum of both file and dir caches sorted and then truncated to the numEntries? I think the overall code for `listStatusFSO` needs to be revisited.